### ISSUE TITLE
imv (the image viewer) is not following the center window rule

### ISSUE THREAD

**Issue by helixoid** [hyprwm/Hyprland#2807]
### Hyprland Version

Hyprland, built from branch  at commit b08b72358ad549fd066e5be0fc3aa4c9df367607 dirty (). Tag: v0.27.2  flags: (if any)

### Bug or Regression?

Bug

### Description

I use imv as the image viewer. I wanted to make it float at the center of the display but it is not following the center window rule.
I also checked the move [x] [y] rule but it seems to work as expected.
I also tried swayimg but it's every instance has different class, so I can't define window rule.

### How to reproduce

```
windowrulev2 = float, class:^(imv)$
windowrulev2 = size 900 600, class:^(imv)$
windowrulev2 = center, class:^(imv)$
```

**Comment by MightyPlaza** [comment]
(imv is looking for main dev)

setting size using compositor seems to break imv (likely https://todo.sr.ht/~exec64/imv/47), so simply use the imv config file

doing this makes the window have the intended size and the center window rule now works as well

this being said, there doesn't seem to be much that can be done on Hyprland's side

**Comment by vaxerski** [comment]
like?

**Comment by MightyPlaza** [comment]
I'm working on it dw
it happens when size rules are set before the center rule

**Comment by helixoid** [comment]
> it happens when size rules are set after the center rule

It fixed my issue but it is a workaround should be mentioned on wiki or be fixed.

**Comment by MightyPlaza** [comment]
fixed 901236a5356a3f249af7d55e5db69889e50256b8

**Comment by helixoid** [comment]
Thanx
